src/js/common/schemaform/FormIntroButtons.jsx

```diff
@@ -4,66 +4,34 @@ import PropTypes from 'prop-types';
 import ProgressButton from '../../common/components/form-elements/ProgressButton';
 
 class FormIntroButtons extends React.Component {
-  componentWillMount() {
-    // TODO: When the api is available to get a list of all the saved forms, query
-    //  for that instead and save it in the redux store
-
-    // Load the form data into the redux store if possible
-    // NOTE: This does not fill the data into the form yet; just grabs it and
-    //  saves it for later (when we want to fill in the blanks).
-    // TODO: Probably should wrap this in
-    // if (this.props.loggedIn) { ... }
-    this.props.fetchInProgressForm(this.props.form.formId, this.props.form.migrations);
-  }
-
-  // Load the form data again if we weren't logged in before but are now.
-  componentWillReceiveProps(newProps) {
-    // Ugh. So verbose.
-    let wasLoggedIn;
-    let isLoggedIn;
-    try {
-      wasLoggedIn = this.props.loggedIn;
-    } catch (e) {
-      if (e instanceof TypeError) {
-        // Probably couldn't read a property somewhere along the way
-        wasLoggedIn = false;
-      }
-    }
-    try {
-      isLoggedIn = newProps.loggedIn;
-    } catch (e) {
-      if (e instanceof TypeError) {
-        isLoggedIn = false;
-      }
-    }
-    if (!wasLoggedIn && isLoggedIn) {
-      this.props.fetchInProgressForm(this.props.form.formId, this.props.form.migrations);
-    }
-  }
-
   goToBeginning = () => {
     this.props.router.push(this.props.route.pageList[1].path);
   }
 
   handleLoadForm = () => {
     // This is where we take the formData we pre-loaded in componentWillMount
     //  and fill in the form.
-    this.props.loadInProgressDataIntoForm();
-    // Navigate to the last page they were on
-    this.props.router.push(this.props.form.loadedData.metadata.returnUrl);
-    // TODO: Handle this scenario:
-    //  1) I fill out some information and save my progress.
-    //  2) The form is updated and a field I've not filled out yet gets moved
-    //     to a page I have already completed.
-    //  3) I load my saved progress.
-    //  4) I should be put in the page with the missing information.
+    this.props.fetchInProgressForm(this.props.formId, this.props.migrations).then((returnUrl) => {
```

Comment:
I think it would be better to handle this in a lifecycle method, rather than using the promise. We're setting the correct statuses already and we probably want to make `fetchInProgressForm` work like `saveInProgressForm`, where we don't re-reject in the `catch` callback (in my outstanding PR). Plus, you should have the updated props with the metadata in that case.